src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const changesArray = [];
+
+  for (const deed of actions) {
+    switch (deed.type) {
```

Comment:
You can destructure all the needed props of action (type, keysToRemove, extraData)